test/e2e/scheduling/predicates.go

```diff
@@ -223,6 +222,117 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		verifyResult(cs, podsNeededForSaturation, 1, ns)
 	})
 
+	// This test verifies we don't allow scheduling of pods in a way that sum of local storage limits of pods is greater than machines capacity.
+	// It assumes that cluster add-on pods stay stable and cannot be run in parallel with any other test that touches Nodes or Pods.
+	// It is so because we need to have precise control on what's running in the cluster.
+	It("validates local storage resource limits of pods that are allowed to run [Conformance]", func() {
+		nodeScratchMaxAllocatable := int64(0)
+		nodeOverlayMaxAllocatable := int64(0)
+
+		nodeToScratchAllocatableMap := make(map[string]int64)
+		nodeToOverlayAllocatableMap := make(map[string]int64)
+		for _, node := range nodeList.Items {
+			scratchAllocatable, found := node.Status.Allocatable["storage.kubernetes.io/scratch"]
+			Expect(found).To(Equal(true))
+			if nodeScratchMaxAllocatable < scratchAllocatable.MilliValue() {
+				nodeScratchMaxAllocatable = scratchAllocatable.MilliValue()
+			}
+			nodeToScratchAllocatableMap[node.Name] = scratchAllocatable.MilliValue()
+
+			overlayAllocatable, found := node.Status.Allocatable["storage.kubernetes.io/overlay"]
+			if found {
+				if nodeOverlayMaxAllocatable < overlayAllocatable.MilliValue() {
+					nodeOverlayMaxAllocatable = overlayAllocatable.MilliValue()
+				}
+				nodeToOverlayAllocatableMap[node.Name] = overlayAllocatable.MilliValue()
+			} else {
+				nodeToOverlayAllocatableMap[node.Name] = 0
+			}
+		}
+		framework.WaitForStableCluster(cs, masterNodes)
+
+		pods, err := cs.Core().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})
+		framework.ExpectNoError(err)
+		for _, pod := range pods.Items {
+			_, found := nodeToScratchAllocatableMap[pod.Spec.NodeName]
+			if found && pod.Status.Phase != v1.PodSucceeded && pod.Status.Phase != v1.PodFailed {
+				scratchRequested := getRequestedStorageScratch(pod)
+				overlayRequested := getRequestedStorageOverlay(pod)
+				framework.Logf("Pod %v requesting resource scratch=%v , overlay=%v on Node %v", pod.Name, scratchRequested, overlayRequested, pod.Spec.NodeName)
+				if nodeToOverlayAllocatableMap[pod.Spec.NodeName]/overlayRequested == 0 {
+					nodeToScratchAllocatableMap[pod.Spec.NodeName] = nodeToScratchAllocatableMap[pod.Spec.NodeName] - scratchRequested - overlayRequested
```

Comment:
I think the logic is 
1. One partition, container overlay consumes node scratch space
2. Two partition, container overly consumes node overlay resource

I don't see a condition that a pod request that if is not satisfied by overlay, it can satisfy by scratch space?